### PR TITLE
Add sequential sampling mode with configurable overlap

### LINKED ISSUE #2: Add more Data Sampling Strategies
The current implementation randomly samples crops from a big volume or bounding box. There should be more data sampling strategies, such as sequential sampling etc. This is low priority for now.

### PR DESCRIPTION
**Pull request by dchen116**
Closes #2

## What

Adds `sampling="sequential"` to `MiaoConfig` for deterministic grid-based iteration over entire volumes — needed for dense inference and evaluation (e.g. running a linear probe over a full EM volume to get quantitative boundary detection metrics).

## Changes

- **`config.py`**: new `sampling` (`"random"` | `"sequential"`) and `overlap` (int or per-axis list) fields with validation
- **`dataset.py`**: `_build_sequential_grid()` precomputes all `(vol_idx, center, grid_index)` positions at init time; `__len__`, `__getitem__`, and `_print_summary` branch on sampling mode; `meta["grid_index"]` tuple added in sequential mode for patch stitching
- **`tests/test_dataset.py`**: 11 new test cases (grid size, full coverage, determinism, overlap, multi-volume, per-axis overlap, error cases)
- **`README.md`**: documents new config fields and sequential usage example
- **`pyproject.toml`**: adds pixi environment config for development (`pixi run test`)

## Usage

\`\`\`yaml
sampling: "sequential"
overlap: 16            # or per-axis list e.g. [16, 16, 8]
\`\`\`

\`\`\`python
dataset = VolumeDataset(config)   # len = total grid positions
loader = DataLoader(dataset, shuffle=False)

for batch in loader:
    grid_index = batch["meta"]["grid_index"]  # use to stitch predictions
\`\`\`

### CODE REVIEW

**Comment by jakobtroidl** [comment]
@dchen116, could you please add a note to the README that expains how `isotropic` and `sequential sampling` interact. Otherwise, looks good on my end.
